agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/common/NetworkFriendlyExceptions.java

```diff
@@ -49,9 +56,46 @@ public static boolean logSpecialOneTimeFriendlyException(
       //  e.g. wrong host address or cannot reach address due to network issues...
       return false;
     }
+
+    IOException ioException = getCausedByOfType(error, IOException.class);
+    if (ioException != null) {
+      if (!alreadySeen.getAndSet(true)) {
+        List<String> missingCiphers;
+        try {
+          missingCiphers = getMissingCiphers();
+        } catch (NoSuchAlgorithmException e) {
+          logger.error(e.getMessage(), e);
+          return false;
+        }
+        if (missingCiphers != null && missingCiphers.size() > 0) {
+          logger.error(getCipherFriendlyMessage(url, missingCiphers));
+        } else {
+          return false;
+        }
+      }
+      return true;
+    }
     return false;
   }
 
+  private static List<String> getMissingCiphers() throws NoSuchAlgorithmException {
+    final List<String> missingCiphers = new ArrayList<>();
+    final List<String> expectedCiphers =
+        Arrays.asList(
+            "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
+            "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
+            "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384",
+            "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256");
+    SSLSocketFactory socketFactory = SSLContext.getDefault().getSocketFactory();
+    List<String> cipherSuitesFromJvm = Arrays.asList(socketFactory.getSupportedCipherSuites());
+    for (String cipher : expectedCiphers) {
+      if (!cipherSuitesFromJvm.contains(cipher)) {
+        missingCiphers.add(cipher);
+      }
+    }
+    return missingCiphers;
```

Comment:
since it's only a problem if missing all of the above ciphers, what do u think about only returning here if it's all of them?